### PR TITLE
[MI-3812]: Added the test cases for UI of list of linked channels in RHS on MM.

### PR DESCRIPTION
**Pull request by arush-vashishtha**
### Summary
This PR consists the test cases for the UI of list of linked channels in RHS on MM.

### CODE REVIEW

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/List_linked_channels.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+# (Required) Ensure all values are filled up
+name: "List of linked channels in RHS"
+status: Active
+priority: Normal
+folder: UI-design-1.0
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/List_linked_channels.md]: Fix the folder name everywhere.

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Scrolling_in_linked_channels_list.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+---
+# (Required) Ensure all values are filled up
+name: "Scrolling in the list of linked channels in RHS"
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Scrolling_in_linked_channels_list.md]:
```suggestion
name: "Infinite scroll in the list of linked channels in RHS"
```

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Scrolling_in_linked_channels_list.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Scroll down and up in the list of linked channels in the RHS of MS Teams sync plugin on MM.
+
+**Expected**
+
+The user should be able to scroll down and up in the list of linked channels in the RHS of MS Teams sync plugin on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Scrolling_in_linked_channels_list.md]:
```suggestion
The user should be able to scroll down and up in the list of linked channels and verify all the channels linked are present in the RHS of MS Teams sync plugin on MM.
```

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Search_in_linked_channels_list.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Enter linked channel name in the search bar above the list of linked channels in the RHS of MS Teams sync plugin on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Search_in_linked_channels_list.md]:
```suggestion
1. Enter linked channel name in the search bar present above the list of linked channels in the RHS of MS Teams sync plugin on MM.
```

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Search_in_linked_channels_list.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+**Step 2**
+
+1. Enter unlinked channel name on invalid channel name in the search bar above the list of linked channels in the RHS of MS Teams sync plugin on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Search_in_linked_channels_list.md]:
```suggestion
1. Enter unlinked channel name or invalid channel name in the search bar present above the list of linked channels in the RHS of MS Teams sync plugin on MM.
```

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Slash_commands_linked_channel_list.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Enter the slash command `/msteams link [msteams-teamID] [msteams-channelID]` in the unlinked channel on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Slash_commands_linked_channel_list.md]: write the correct slash command everywhere, also put the variables under <>.

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Slash_commands_linked_channel_list.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+2. Click on MS Teams sync plugin icon in App bar or channel header and open the RHS on MM.
+
+**Step 2**
+
+1. Enter the slash command `/msteams unlink` in the linked channel on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Slash_commands_linked_channel_list.md]: After step 1, because we need to have atleast 1 channel linked

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Slash_commands_linked_channel_list.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+**Step 3**
+
+1. Enter the slash command `/msteams unlink` in all the linked channels on MM
+2. Click on MS Teams sync plugin icon in App bar or channel header and open the RHS on MM.
+
+**Expected**
+
+The user should be able to verify the linked channel in the list of linked channels in the RHS of MS Teams sync plugin on MM.
+After step 2, The unlinked channel should not be present in the list of linked channels in the RHS of MS Teams sync plugin on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Slash_commands_linked_channel_list.md]:
```suggestion
After step 2, the desired channel unlinked should not be present in the list of linked channels in the RHS of MS Teams sync plugin on MM.
```

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Slash_commands_linked_channel_list.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+After step 3, The user should be able to see the `Link a channel` button in the RHS of MS Teams sync plugin on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Slash_commands_linked_channel_list.md]:
```suggestion
After step 3, the user should be able to see the `Link a channel` button in the RHS of MS Teams sync plugin on MM.
```

Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Slash_commands_linked_channel_list.md]: Rephrase this as discussed.

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Unlink_all_channel_RHS.md [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Unlink all the linked channels from the list of linked channels on MM from RHS of MS Teams sync plugin.
+
+**Expected**
+
+The user should be able to see the `Link a channel` button in the RHS of MS Teams sync plugin on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Unlink_all_channel_RHS.md]: Please add the list statement.

File: data/test-cases/plugins/ms-teams-sync/ui-integration-phase-2/List_linked_channels.md [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Click on MS Teams sync icon in App bar or channel header and open the RHS when there are linked channels on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/ui-integration-phase-2/List_linked_channels.md]:
```suggestion
1. Click on MS Teams sync icon in App bar or channel header and open the RHS when there are MM channel linked to MS teams.
```

File: data/test-cases/plugins/ms-teams-sync/ui-integration-phase-2/Slash_commands_linked_channel_list.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+2. Click on MS Teams sync plugin icon in App bar or channel header and open the RHS on MM.
+
+**Step 3**
+
+1. Enter the slash command `/msteams unlink` in all the linked channels on MM
+2. Click on MS Teams sync plugin icon in App bar or channel header and open the RHS on MM.
+
+**Expected**
+
+The user should be able to verify the linked channel in the `list of linked channels` in the RHS of MS Teams sync plugin on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/ui-integration-phase-2/Slash_commands_linked_channel_list.md]:
```suggestion
The user should be able to verify the newly linked channel in the `list of linked channels` in the RHS of MS Teams sync plugin on MM.
```

**Review by AayushChaudhary0001 — APPROVED** [review]

File: data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Click on MS Teams sync icon in App bar or channel header and open the RHS when there are MM channels linked to MS Teams channel.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md]:
```suggestion
1. Click on the MS Teams sync icon in the app bar or channel header to open the RHS when there are MM channels linked to the MS Teams channel.
```

Comment (ayusht2810) [reply data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md]: fix everywhere

File: data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+
+**Step 2**
+
+1. Click on MS Teams sync icon in App bar or channel header and open the RHS when there are MM channels linked to MS Teams channel and the logged-in user is a `normal user`.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md]: I don't think normal user is the correct term here. We should update this to something else.

File: data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+
+**Step 3**
+
+1. Click on MS Teams sync icon in App bar or channel header and open the RHS when there are MM channels linked to MS Teams channel and the logged-in user is a `system admin`.
+
+**Expected**
+
+The user should be able to see the `list of linked channels` in the RHS of MS Teams sync plugin on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md]: What is the use of this test case?

File: data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+After step 2, the user should be able to see those linked channels on MM in the `list of linked channels` in the RHS of MS Teams sync plugin in which they are added as a member on MM.
+After step 3, the user should be able to see all the linked channels on MM in the `list of linked channels` in the RHS of MS Teams sync plugin on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md]:
```suggestion
After step 2, the user should be able to see the list of linked channels in the RHS of the MS Teams sync plugin in which they are added as a member in MM.
After step 3, the user should be able to see the list of all the linked channels in the RHS of the MS Teams sync plugin on MM.
```

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Scrolling_in_linked_channels_list.md [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Scroll down and up in the `list of linked channels` in the RHS of MS Teams sync plugin on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Scrolling_in_linked_channels_list.md]: why down and up?

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Scrolling_in_linked_channels_list.md [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+
+**Expected**
+
+The user should be able to scroll down and up in the `list of linked channels` and verify all the linked channels are present in list in the RHS of MS Teams sync plugin on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Scrolling_in_linked_channels_list.md]: same. also should we write the text present at the end of the list?

File: data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md (hunk 16 above)
Comment (ayusht2810) [reply data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md]: fix everywhere

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Search_in_linked_channels_list.md [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Enter `linked channel` name in the search bar present above the `list of linked channels` in the RHS of MS Teams sync plugin on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Search_in_linked_channels_list.md]:
```suggestion
1. Enter the `mattermost linked channel` name in the search bar present above the `list of linked channels` in the RHS of the MS Teams sync plugin on MM.
```

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Search_in_linked_channels_list.md [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+
+**Step 2**
+
+1. Enter `unlinked channel` name or `invalid channel` name in the search bar present above the `list of linked channels` in the RHS of MS Teams sync plugin on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Search_in_linked_channels_list.md]:
```suggestion
1. Enter the `unlinked channel` name or `invalid channel` name in the search bar present above the `list of linked channels` in the RHS of the MS Teams sync plugin on MM.
```

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Search_in_linked_channels_list.md [hunk 21]
```diff
@@ -0,0 +1,46 @@
...
+
+**Expected**
+
+The user should be able to verify the searched linked channel in the `list of linked channels` in the RHS of MS Teams sync plugin on MM.
+After step 2, The user should get the `search not found` message in the RHS of MS Teams sync plugin on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Search_in_linked_channels_list.md]:
```suggestion
The user should be able to verify the searched linked channel in the `list of linked channels` in the RHS of the MS Teams sync plugin on MM.
After step 2, the user should get the `search not found` message in the RHS of the MS Teams sync plugin on MM.
```

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Slash_commands_linked_channel_list.md [hunk 22]
```diff
@@ -0,0 +1,54 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Enter the slash command `/msteams-sync link <msteams-teamID> <msteams-channelID>` in the unlinked channel on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Slash_commands_linked_channel_list.md]:
```suggestion
1. Link an unlinked channel by running the slash command `/msteams-sync link <msteams-teamID> <msteams-channelID>` on MM.
```

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Tool_tip_linked_channel_list.md [hunk 23]
```diff
@@ -0,0 +1,41 @@
+---
+# (Required) Ensure all values are filled up
+name: "Tool tip upon hovering over the name of linked channel"
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Tool_tip_linked_channel_list.md]:
```suggestion
name: "Tooltip on hovering over the linked channel details"
```

Comment (ayusht2810) [reply data/test-cases/plugins/ms-teams-sync/ui-integration/Tool_tip_linked_channel_list.md]: tooltip is a single word. fix everywhere

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Tool_tip_linked_channel_list.md [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Hover over the channel name in the `list of linked channels` in the RHS of MS Teams sync plugin on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Tool_tip_linked_channel_list.md]: Should we mention the team name as well?

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Unlink_all_channel_RHS.md [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Unlink all the linked channels from the `list of linked channels` on MM from RHS of MS Teams sync plugin.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Unlink_all_channel_RHS.md]: Isn't this test case part of the unlink feature?

File: data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md [hunk 26]
```diff
@@ -0,0 +1,51 @@
+---
+# (Required) Ensure all values are filled up
+name: "List of linked channels in RHS"
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md:3]: Do we need to add a test case when no linked channels are present?

File: data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md (hunk 14 above)
Comment (arush-vashishtha) [reply data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md]: then i think we should update it as member

File: data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md (hunk 15 above)
Comment (arush-vashishtha) [reply data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md]: done removed.

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Unlink_all_channel_RHS.md (hunk 25 above)
Comment (arush-vashishtha) [reply data/test-cases/plugins/ms-teams-sync/ui-integration/Unlink_all_channel_RHS.md]: added in the phase 3 cases

File: data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md (hunk 26 above)
Comment (arush-vashishtha) [reply data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md:3]: added a scenario

File: data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md [hunk 27]
```diff
@@ -26,26 +26,26 @@ manual_test_environments: []
 # Do not change
 id: null
 key: null
-created_on: null
+created_on: nullthe `list of linked channels` should be empty and the user should be able to see the `Link a channel` button in the RHS of MS Teams sync plugin on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md]:
```suggestion
created_on: null
```

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Slash_commands_linked_channel_list.md [hunk 28]
```diff
@@ -34,21 +34,21 @@ steps_hashed: null
...
 
-1. Enter the slash command `/msteams-sync unlink` in the linked channel on MM.
+1. Unlink an linked channel by running the slash command `/msteams-sync unlink` on MM.
 2. Click on MS Teams sync plugin icon in App bar or channel header and open the RHS on MM.
 
 **Step 3**
 
-1. Enter the slash command `/msteams unlink` in all the linked channels on MM
+1. Unlink all linked channels by running the slash command `/msteams-sync unlink` in all linked channels on MM.
 2. Click on MS Teams sync plugin icon in App bar or channel header and open the RHS on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Slash_commands_linked_channel_list.md]:
```suggestion
2. Click on the MS Teams sync plugin icon in the app bar or channel header and open the RHS on MM.
```

Comment (ayusht2810) [reply data/test-cases/plugins/ms-teams-sync/ui-integration/Slash_commands_linked_channel_list.md]: fix everywhere

File: data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md (hunk 14 above)
Comment (ayusht2810) [reply data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md]: i think we can use a term something like member having admin permissions and member having non-admin permissions

File: data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md [hunk 29]
```diff
@@ -0,0 +1,51 @@
...
+
+**Step 3**
+
+1. Click on the MS Teams sync icon in the app bar or channel header to open the RHS when there are no MM channels linked to the MS Teams channel.
+
+**Expected**
+
+The user should be able to see the list of linked channels in the RHS of the MS Teams sync plugin in which they are added as a member in MM.
+After step 2, the user should be able to see the list of all the linked channels in the RHS of the MS Teams sync plugin on MM.
+After step 3, the `list of linked channels` should be empty and the user should be able to see the `Link a channel` button in the RHS of the MS Teams sync plugin on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md]: add a test case of linked channels visible when user is disconnected

**Review by ayusht2810 — APPROVED** [review]

File: data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md [hunk 30]
```diff
@@ -44,8 +44,13 @@ steps_hashed: null
...
+2. Click on the MS Teams sync icon in the app bar or channel header to open the RHS when there are MM channels linked to the MS Teams channel
+
 **Expected**
 
 The user should be able to see the list of linked channels in the RHS of the MS Teams sync plugin in which they are added as a member in MM.
 After step 2, the user should be able to see the list of all the linked channels in the RHS of the MS Teams sync plugin on MM.
-After step 3, the `list of linked channels` should be empty and the user should be able to see the `Link a channel` button in the RHS of the MS Teams sync plugin on MM.
\ No newline at end of file
+After step 3, the `List of linked channels` should be empty and the user should be able to see the `Link a channel` button in the RHS of the MS Teams sync plugin on MM.
+After step 4, the user should be able to see the `Request message` to connect their MS Teams account along with the `List of linked channels` in the RHS and the toast message regarding the successful disconnection of their MS Teams account.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/List_linked_channels.md]:
```suggestion
After step 3, the `list of linked channels` should be empty and the user should be able to see the `Link a channel` button in the RHS of the MS Teams sync plugin on MM.
After step 4, the user should be able to see the `Request message` to connect their MS Teams account along with the `list of linked channels` in the RHS and the toast message regarding the successful disconnection of their MS Teams account.
```

**Review by ayusht2810 — APPROVED** [review]
